master/internal/workspace/postgres_workspace_intg_test.go

```diff
@@ -0,0 +1,136 @@
+//go:build integration
+// +build integration
+
+package workspace
+
+import (
+	"context"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/uptrace/bun"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/etc"
+	"github.com/determined-ai/determined/proto/pkg/workspacev1"
+)
+
+const (
+	cluster1 = "C1"
+	cluster2 = "C2"
+)
+
+func TestMain(m *testing.M) {
+	pgDB, err := db.ResolveTestPostgres()
+	if err != nil {
+		log.Panicln(err)
+	}
+
+	err = db.MigrateTestPostgres(pgDB, "file://../../static/migrations", "up")
+	if err != nil {
+		log.Panicln(err)
+	}
+
+	err = etc.SetRootPath("../../static/srv")
+	if err != nil {
+		log.Panicln(err)
+	}
+
+	os.Exit(m.Run())
+}
+
+type Bindings struct {
+	bun.BaseModel `bun:"table:workspace_namespace_bindings"`
+	WorkspaceID   int    `bun:"workspace_id"`
+	ClusterName   string `bun:"cluster_name"`
+	NamespaceName string `bun:"namespace_name"`
+}
+
+func TestGetNamespaceFromWorkspace(t *testing.T) {
+	ctx := context.Background()
+	wksp1 := "wksp_1"
+	wkspID1, _ := db.RequireMockWorkspaceID(t, db.SingleDB(), wksp1)
+	wksp2 := "wksp_2"
+	wkspID2, _ := db.RequireMockWorkspaceID(t, db.SingleDB(), wksp2)
+	wksp3 := "wksp_3"
+	wkspID3, _ := db.RequireMockWorkspaceID(t, db.SingleDB(), wksp3)
+	cluster1 := "C1"
+	cluster2 := "C2"
+
+	b1 := Bindings{WorkspaceID: wkspID1, ClusterName: cluster1, NamespaceName: "n1"}
+	b2 := Bindings{WorkspaceID: wkspID1, ClusterName: cluster2, NamespaceName: "n2"}
+	b3 := Bindings{WorkspaceID: wkspID2, ClusterName: cluster1, NamespaceName: "n3"}
```

Comment:
write some creative tests. what happens when you reinsert the same binding? what happens with queries after that, etc.

Reply:
I think reinserting the same binding would be out of scope for this test, because it is testing if we can get the namespace given the workspace and cluster name. It should be a test for a function that would insert bindings (probably created in Amanda's PR) and that test should result in an error after the PRIMARY key addition.

I have added a few more tests that look for a non-existent workspace id, a non-existent cluster name, and both at the same time.

Reply:
My point was mostly that if you did write this test, you'd just be testing your data model and you'd probably learn that duplicate entries were allowed without the primary key change.